internal/operators/cnv/cnv_operator.go

```diff
@@ -91,10 +91,12 @@ func (o *operator) ValidateHost(ctx context.Context, cluster *common.Cluster, ho
 		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{"CPU does not have virtualization support "}}, nil
 	}
 
+	role := common.GetEffectiveRole(host)
 	// If the Role is set to Auto-assign for a host, it is not possible to determine whether the node will end up as a master or worker node.
-	if host.Role == models.HostRoleAutoAssign {
-		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{"All host roles must be assigned to enable CNV"}}, nil
+	if role == models.HostRoleAutoAssign {
```

Comment:
Please add a test for changes in this file.

Reply:
There was already test in place for this line and I changed the expected value to Pending. Semantically, failure and pending are the same BTW so overall the behavior did not changed